### ISSUE TITLE
Getting Exception in thread “main” java.lang.ArrayIndexOutOfBoundsException while retraining the stanford sentiment RNN

### ISSUE THREAD

**Issue by ShubhankarKothari** [stanfordnlp/CoreNLP#499]
I am trying to retrain the Stanford sentiment RNN with my own dataset which I have converted into PTB format using BuildBinarizedDataset from the Stanford-core-nlp jar (3.8) 

Here are the first few lines of my training data:

(9 (9 (5 very) (7 well)) (5 handled))

(7 (6 engineers) (4 (4 to) (5 (6 answer) (6 (5 very) (3 carefully)))))

(9 (9 (9 (8 fast) (5 and)) (9 reliable)) (9 (4 response) (9 great)))

(8 (7 prompt) (7 (5 responses) (8 (5 fixed) (1 (4 the) (0 issue)))))

This is the command I used :

java -cp "*" 

-mx8g edu.stanford.nlp.sentiment.SentimentTraining 

-gradientcheck 

-trainPath train_s.txt -equivalenceClasses 0,1,2,3,4,5;6,7,8,9,10 

-equivalenceClassNames Negative,Neutral,Positive

-numHid 25 

-batchSize 78 

-numClasses 2 

-classNames Negative,Positive 

-train 

-model modeltrial.ser.gz

And this is my exception:

Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 9
   at org.ejml.data.D1Matrix64F.set(Unknown Source)

   at org.ejml.simple.SimpleBase.set(Unknown Source)

   at edu.stanford.nlp.sentiment.SentimentCostAndGradient.backpropDerivativesAndError(SentimentCostAndGradient.java:376)

   at edu.stanford.nlp.sentiment.SentimentCostAndGradient.backpropDerivativesAndError(SentimentCostAndGradient.java:354)

   at edu.stanford.nlp.sentiment.SentimentCostAndGradient.scoreDerivatives(SentimentCostAndGradient.java:223)

   at edu.stanford.nlp.sentiment.SentimentCostAndGradient.calculate(SentimentCostAndGradient.java:249)

   at edu.stanford.nlp.optimization.AbstractCachingDiffFunction.ensure(AbstractCachingDiffFunction.java:140)

   at edu.stanford.nlp.optimization.AbstractCachingDiffFunction.derivativeAt(AbstractCachingDiffFunction.java:151)

   at edu.stanford.nlp.optimization.AbstractCachingDiffFunction.gradientCheck(AbstractCachingDiffFunction.java:39)

   at edu.stanford.nlp.sentiment.SentimentTraining.runGradientCheck(SentimentTraining.java:129)

   at edu.stanford.nlp.sentiment.SentimentTraining.main(SentimentTraining.java:226)

These are the options I am working with

GENERAL OPTIONS

randomSeed=1659449035

wordVectors=null

unkWord=UNK

randomWordVectors=true

numHid=25

numClasses=2

lowercaseWordVectors=false

useTensors=true

simplifiedModel=true

combineClassification=true

classNames=Negative,Positive

equivalenceClasses=0,1,2,3,4,5;6,7,8,9,10

equivalenceClassNames=Negative,Neutral,Positive

TRAIN OPTIONS

batchSize=78

epochs=400

debugOutputEpochs=8

maxTrainTimeSeconds=86400

learningRate=0.01

scalingForInit=1.0

classWeights=null

regTransformMatrix=0.001

regTransformTensor=0.001

regClassification=1.0E-4

regWordVector=1.0E-4

initialAdagradWeight=0.0

adagradResetFrequency=1

shuffleMatrices=true

initialMatrixLogPath=null

nThreads=1

TEST OPTIONS

ngramRecordSize=0

ngramRecordMaximumLength=0

printLengthAccuracies=false

My training data has around 4800 sentences. This is the first time I am working with Stanford core NLP and so far no other links have helped me.
 
stack overflow link : https://stackoverflow.com/questions/45322871/getting-exception-in-thread-main-java-lang-arrayindexoutofboundsexception-whil

**Comment by J38** [comment]
I think you need to set it to `-numClasses 11`
